### PR TITLE
Bug 1836167 - New telemetry in bookmarks for growth retention measurement

### PR DESCRIPTION
**Pull request by HarrisonOg**
For [#1836167](https://bugzilla.mozilla.org/show_bug.cgi?id=1836167)

### Pull Request checklist

- [x] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1836167

### CODE REVIEW

**Comment by HarrisonOg** [comment]
!! Reminder: it is your responsibility to complete and check the correctness of
!! this automatically-generated request skeleton before requesting Data
!! Collection Review. See https://wiki.mozilla.org/Data_Collection for details.

DATA REVIEW REQUEST
1. What questions will you answer with this data?

- This data will help us understand how users are interacting with bookmarks.

2. Why does Mozilla need to answer these questions? Are there benefits for users?
   Do we need this information to address product or business requirements?

- We are implementing more telemetry to provide greater insight into our growth marketing and retention of new users.  This data will help us establish a baseline usage rate for bookmarks so we can better focus our attention on widely used features.

3. What alternative methods did you consider to answer these questions?
   Why were they not sufficient?

- There are no alternatives, we do not have the telemetry to measure bookmark usage at this point.

4. Can current instrumentation answer these questions?

- No, there is no current instrumentation to measure bookmarks.

5. List all proposed measurements and indicate the category of data collection for each
   measurement, using the Firefox data collection categories found on the Mozilla wiki.

Measurement Name | Measurement Description | Data Collection Category | Tracking Bug
---------------- | ----------------------- | ------------------------ | ------------
`metrics.bookmarks_add` | A counter that indicates how many bookmarks a user has added.  The label for this counter is `<source>`.  `source` will be: `page_action_menu` as that is the only entry point right now to add bookmarks.  | interaction | https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
`metrics.bookmarks_delete` | A counter that indicates how many bookmarks a user has deleted.  The label for this counter is `<source>`.  `source` will be: `add_bookmark_toast` or `bookmark_panel`.  | interaction | https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
`metrics.bookmarks_edit` | A counter that indicates how many bookmarks a user has edited.  The label for this counter is `<source>`.  `source` will be: `bookmark_edit_page` or `bookmark_panel`.  | interaction | https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
`metrics.bookmarks_open` | A counter that indicates how many bookmarks a user has opened.  The label for this counter is `<source>`.  `source` will be: `top_sites`, `awesomebar_results`, `bookmark_panel`.  | interaction | https://bugzilla.mozilla.org/show_bug.cgi?id=1836167

6. Please provide a link to the documentation for this data collection which
   describes the ultimate data set in a public, complete, and accurate way.

This collection is Glean so is documented [in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).

7. How long will this data be collected?

This collection will be collected permanently.
cgordon@mozilla.com, rzhao@mozilla.com, android-probes@mozilla.com will be responsible for the permanent collections.

8. What populations will you measure?

All channels, countries, and locales. No filters.

9. If this data collection is default on, what is the opt-out mechanism for users?

These collections are Glean. The opt-out can be found in the product's preferences.

10. Please provide a general description of how you will analyze this data.

- Glean and Amplitude

11. Where do you intend to share the results of your analysis?

- Glean, Amplitude and with mobile teams.

12. Is there a third-party tool (i.e. not Glean or Telemetry) that you
    are proposing to use for this data collection?

No.

**Comment by rocketsroger** [comment]
> !! Reminder: it is your responsibility to complete and check the correctness of !! this automatically-generated request skeleton before requesting Data !! Collection Review. See https://wiki.mozilla.org/Data_Collection for details.
> 
> DATA REVIEW REQUEST
> 
>     1. What questions will you answer with this data?
> 
> 
>     * This data will help us understand how users are interacting with bookmarks.
> 
> 
>     2. Why does Mozilla need to answer these questions? Are there benefits for users?
>        Do we need this information to address product or business requirements?
> 
> 
>     * We are implementing more telemetry to provide greater insight into our growth marketing and retention of new users.  This data will help us establish a baseline usage rate for bookmarks so we can better focus our attention on widely used features.
> 
> 
>     3. What alternative methods did you consider to answer these questions?
>        Why were they not sufficient?
> 
> 
>     * There are no alternatives, we do not have the telemetry to measure bookmark usage at this point.
> 
> 
>     4. Can current instrumentation answer these questions?
> 
> 
>     * No, there is no current instrumentation to measure bookmarks.
> 
> 
>     5. List all proposed measurements and indicate the category of data collection for each
>        measurement, using the Firefox data collection categories found on the Mozilla wiki.
> 
> 
> Measurement Name 	Measurement Description 	Data Collection Category 	Tracking Bug
> `metrics.bookmarks_add` 	A counter that indicates how many bookmarks a user has added. The label for this counter is `<source>`. `source` will be: `page_action_menu` as that is the only entry point right now to add bookmarks. 	interaction 	https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
> `metrics.bookmarks_delete` 	A counter that indicates how many bookmarks a user has deleted. The label for this counter is `<source>`. `source` will be: `add_bookmark_toast` or `bookmark_panel`. 	interaction 	https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
> `metrics.bookmarks_edit` 	A counter that indicates how many bookmarks a user has edited. The label for this counter is `<source>`. `source` will be: `bookmark_edit_page` or `bookmark_panel`. 	interaction 	https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
> `metrics.bookmarks_open` 	A counter that indicates how many bookmarks a user has opened. The label for this counter is `<source>`. `source` will be: `top_sites`, `awesomebar_results`, `bookmark_panel`. 	interaction 	https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
> 
>     6. Please provide a link to the documentation for this data collection which
>        describes the ultimate data set in a public, complete, and accurate way.
> 
> 
> This collection is Glean so is documented [in the Glean Dictionary](https://dictionary.telemetry.mozilla.org).
> 
>     7. How long will this data be collected?
> 
> 
> This collection will be collected permanently. [cgordon@mozilla.com](mailto:cgordon@mozilla.com), [rzhao@mozilla.com](mailto:rzhao@mozilla.com), [android-probes@mozilla.com](mailto:android-probes@mozilla.com) will be responsible for the permanent collections.
> 
>     8. What populations will you measure?
> 
> 
> All channels, countries, and locales. No filters.
> 
>     9. If this data collection is default on, what is the opt-out mechanism for users?
> 
> 
> These collections are Glean. The opt-out can be found in the product's preferences.
> 
>     10. Please provide a general description of how you will analyze this data.
> 
> 
>     * Glean and Amplitude
> 
> 
>     11. Where do you intend to share the results of your analysis?
> 
> 
>     * Glean, Amplitude and with mobile teams.
> 
> 
>     12. Is there a third-party tool (i.e. not Glean or Telemetry) that you
>         are proposing to use for this data collection?
> 
> 
> No.

# Data Review

1) Is there or will there be **documentation** that describes the schema for the ultimate data set in a public, complete, and accurate way? 

Yes, through the metrics.yaml file and the [Glean Dictionary](https://dictionary.telemetry.mozilla.org/apps/fenix )

2) Is there a control mechanism that allows the user to turn the data collection on and off? 

Yes, through the "Send Usage Data" preference in the application settings

3) If the request is for permanent data collection, is there someone who will monitor the data over time?

cgordon@mozilla.com will be responsible for monitoring the data over time

4) Using the **[category system of data types](https://wiki.mozilla.org/Data_Collection)** on the Mozilla wiki, what collection type of data do the requested measurements fall under?

Category 2, Interaction data

5) Is the data collection request for default-on or default-off?

default-on

6) Does the instrumentation include the addition of **any *new* identifiers**?

No

7) Is the data collection covered by the existing Firefox privacy notice? 

Yes

8) Does the data collection use a third-party collection tool? 

No

## Result

data-review+

File: fenix/app/metrics.yaml [hunk 1]
```diff
@@ -1607,6 +1607,107 @@ metrics:
...
+    bugs:
+      - https://bugzilla.mozilla.org/show_bug.cgi?id=1836167
+    data_reviews:
+      - https://github.com/mozilla-mobile/firefox-android/pull/2375
+    data_sensitivity:
+      - interaction
+    notification_emails:
+      - android-probes@mozilla.com
+      - cgordon@mozilla.com
+      - rzhao@mozilla.com
```
Comment (rocketsroger) [inline fenix/app/metrics.yaml:1631]: Please remove rzhao@mozilla.com on the notification emails.  This applies to the other 3 as well.  Thanks

**Review by rocketsroger — COMMENTED** [review]
Looks good.

**Review by rocketsroger — APPROVED** [review]
Please remove [rzhao@mozilla.com](mailto:rzhao@mozilla.com) on the notification emails. This applies to the 3 other locations as well. Thanks

**Review by rocketsroger — APPROVED** [review]

**Pull request merged**